cli/src/main.rs

```diff
@@ -472,6 +503,48 @@ async fn start_background(http_port: &u16, rpc_port: &u16) {
         .expect("Unable to spawn development_server");
 }
 
+async fn new() -> Result<(), DynError> {
+    if let Err(e) = try_new().await {
+        error!("{}", e);
+
+        std::process::exit(-1);
+    }
+
+    return Ok(());
+}
+
+async fn try_new() -> Result<(), DynError> {
```

Comment:
could you help me understand what we're doing here and when/where we expect this function to run?

Reply:
`cargo run new` triggers `fn new()` which triggers `fn try_new()`.... when the CLI binaries are published to npm this would be called when calling `mycelia new`